storage/innobase/trx/trx0rec.cc

```diff
@@ -1859,6 +1859,11 @@ trx_undo_report_row_operation(
 		ut_ad(thr->run_node);
 		ut_ad(que_node_get_type(thr->run_node) == QUE_NODE_INSERT);
 		ut_ad(trx->bulk_insert);
+
+		if (m.first->second.has_written_empty_undo()) {
+			m.first->second.write_undo_empty();
```

Comment:
It would seem that these two can be combined in a single member function invocation.